lib/jsonapi/deserialization.rb

```diff
@@ -62,7 +62,7 @@ def jsonapi_deserialize(document, options = {})
 
       relationships.map do |assoc_name, assoc_data|
         assoc_data = (assoc_data || {})['data'] || {}
-        rel_name = jsonapi_inflector.singularize(assoc_name)
+        rel_name = jsonapi_inflector.singularize(assoc_name).underscore
```

Comment:
@kennstenicht we will want to call the `jsonapi_inflector.underscore` separately, otherwise non-rails usage will not be supported.